service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapper.java

```diff
@@ -41,182 +41,244 @@
+                templateParameters.participant(participant);
 
-    private String buildIdentifierDescription(ReferralRequest referralRequest) {
-        Optional<String> identifier = Optional.empty();
-        if (referralRequest.hasIdentifier()) {
-            identifier = referralRequest.getIdentifier()
-                .stream()
-                .filter(val -> val.getSystem().equals(UBRN_SYSTEM_URL))
-                .map(Identifier::getValue)
-                .findFirst();
+            } else if (isReferenceToType(agent, ResourceType.Practitioner) || isReferenceToType(agent, ResourceType.Organization)) {
+                final String participantRef = idMapper.getOrNew(agent);
+                final String participant = participantMapper.mapToParticipant(participantRef, ParticipantType.AUTHOR);
+                templateParameters.participant(participant);
+            }
+
+            if (isReferenceToType(agent, ResourceType.Device)) {
+                messageContext.getInputBundleHolder().getResource(agent.getReferenceElement())
+                    .filter(Device.class::isInstance)
+                    .map(Device.class::cast)
+                    .filter(Device::hasType)
+                    .map(Device::getType)
+                    .flatMap(CodeableConceptMappingUtils::extractTextOrCoding)
+                    .map(REQUESTER_DEVICE::concat)
+                    .ifPresentOrElse(this::prependText, () -> {
+                        throw new EhrMapperException("Could not resolve Device Reference");
+                    });
+
+            } else if (isReferenceToType(agent, ResourceType.Organization)) {
+                messageContext.getInputBundleHolder().getResource(agent.getReferenceElement())
+                    .filter(Organization.class::isInstance)
+                    .map(Organization.class::cast)
+                    .filter(Organization::hasName)
+                    .map(Organization::getName)
+                    .map(REQUESTER_ORG::concat)
+                    .ifPresent(this::prependText);
+
+            } else if (isReferenceToType(agent, ResourceType.Patient)) {
+                messageContext.getInputBundleHolder().getResource(agent.getReferenceElement())
+                    .filter(Patient.class::isInstance)
+                    .map($ -> REQUESTER_PATIENT)
+                    .ifPresent(this::prependText);
+
+            } else if (isReferenceToType(agent, ResourceType.RelatedPerson)) {
+                messageContext.getInputBundleHolder().getResource(agent.getReferenceElement())
+                    .filter(RelatedPerson.class::isInstance)
+                    .map(RelatedPerson.class::cast)
+                    .filter(RelatedPerson::hasName)
+                    .map(RelatedPerson::getNameFirstRep)
+                    .map(RequestStatementExtractor::extractHumanName)
+                    .map(REQUESTER_RELATION::concat)
+                    .ifPresent(this::prependText);
```

Comment:
I believe this exception was there previously:
```suggestion
                    .ifPresentOrElse(this::prependText, () -> {
                        throw new EhrMapperException("Could not resolve RelatedPerson Reference");
                    });
```